Training2/src/src/components/Task.js

```diff
@@ -0,0 +1,80 @@
+import { useCallback, useEffect, useState } from "react";
+import Modal from "./Modal";
+import { isNaN } from "lodash";
+
+const Task = () => {
+  const [inputVal, setInputVal] = useState("");
+  const [isStopCountDown, setIsStopCountDown] = useState(false);
+  const [isValidInput, setIsValidInput] = useState(false);
+  const [count, setCount] = useState(0);
+  const [isShowing, setIsShowing] = useState(false);
+  const [alertMessage, setAlertMessage] = useState("");
+
+  const handleClose = useCallback(() => {
+    setIsShowing(false);
+  }, []);
+
+  const handleInputChange = (e) => {
+    setInputVal(e.target.value);
+  };
+
+  const handleStart = () => {
+    if (!inputVal) {
+      setAlertMessage("Please input a number!!!");
+      setIsShowing(true);
+      return;
+    }
+    if (isNaN(+inputVal)) {
+      setAlertMessage("Invalid input. Must be a number!!!");
+      setIsShowing(true);
+      return;
+    }
+    if (inputVal <= 0) {
+      setAlertMessage("Number must be greater than 0!!!");
+      setIsShowing(true);
+      return;
+    }
+    setIsShowing(false);
+    setIsValidInput(true);
+    setCount(+inputVal);
+    setIsStopCountDown(false);
+  };
+
+  useEffect(() => {
+    if (!isValidInput || isStopCountDown) {
+      return;
+    }
+    const countDown = setInterval(() => {
+      setCount((count) => count - 1);
+    }, 1000);
+    return () => {
+      clearInterval(countDown);
+      setIsValidInput(false);
+    };
+  }, [isValidInput, isStopCountDown]);
```

Comment:
Why do you need to put `isValidInput` into the dependencies array?

Reply:
I first want to trigger useEffect whenever the isValidInput change. But after looking at the code again, I think it is useless now. I forgot to set IsValidInput to false when doing the validation of input.